dnf/base.py

```diff
@@ -217,9 +217,14 @@ def _setup_excludes_includes(self, only_main=False):
                 self.sack.add_excludes(query)
 
     def _setup_modules(self):
-        module_defaults_prioritizer = Modulemd.Prioritizer()
-        # HACK: https://github.com/fedora-modularity/libmodulemd/issues/42
-        module_defaults_prioritizer.add([], 0)
+        try:
+            module_defaults_prioritizer = Modulemd.Prioritizer()
+            # HACK: https://github.com/fedora-modularity/libmodulemd/issues/42
+            module_defaults_prioritizer.add([], 0)
+        except TypeError:
+            # For reason see: https://bugzilla.redhat.com/show_bug.cgi?id=1571081
```

Comment:
I am confused here. The original issue was reported with ``dnf upgrade`` command. The solution would be that modularity setting will be skipped completely, but the work-flow continues, therefore if upgrade was requested, it will be performed without modularity rules with all modularity packages. Additionally user get warning "Run 'dnf update' command, please". And if used does what was requested, he again gets the same recommendation. Or am I completely wrong?

Reply:
Issue is dnf wide, not just `upgrade` command. Modularity setup is skipped and yes it will break modularity, but don't you think that breaking modularity is better than breaking whole dnf, which results into users not being able to do anything with dnf. 

After successful update, user will not get any suggestions about updating.

Reply:
So you suggest leaving dnf backtracing, let someone else fix the problem and leave users without working dnf, very good idea. Problem is not in dnf or libmodulemd, sgallagher or me haven't found the cause, but system update solves the issue.

Idea behind this patch was give users some chance to recover from broken state. Without it they can do nothing (well, they can manually comment the lines of code and do the update, which is basically same thing as this patch does/suggests).